### PR TITLE
feat: TokenAirdrop disable contract airdrops

### LINKED ISSUE #14663: TokenAirdrop disable contract airdrops
As per this discussion https://limechain.slack.com/archives/C048K7ZBS9M/p1722406435947579?thread_ts=1722263267.656899&cid=C048K7ZBS9M we are disabling the airdrop to a contract

### PR DESCRIPTION
**Pull request by vtronkov**
**Description**:
* Add validation to check if we airdrop to a contract without an admin key
* **Bonus refactor:** Move the airdrop max value to the config

**Related issue(s)**:

Fixes #14663

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-node/hedera-config/src/main/java/com/hedera/node/config/data/TokensConfig.java [hunk 1]
```diff
@@ -50,6 +50,8 @@ public record TokensConfig(
                 boolean nftsUseVirtualMerkle,
         @ConfigProperty(value = "maxAllowedAirdrops", defaultValue = "1000000") @NetworkProperty
                 long maxAllowedAirdrops,
+        @ConfigProperty(value = "maxAllowedAirdropTransfers", defaultValue = "10") @NetworkProperty
+                long maxAllowedAirdropTransfers,
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-config/src/main/java/com/hedera/node/config/data/TokensConfig.java:54]: Why do we need this?

Comment (vtronkov) [reply hedera-node/hedera-config/src/main/java/com/hedera/node/config/data/TokensConfig.java:54]: Similar to this discussion https://github.com/hashgraph/hedera-services/pull/14379#discussion_r1697061341 - I am moving this value to a config.

I'm not sure if we want to have 3 different configurations for this or only one. Maybe here is the misunderstanding?

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenAirdropHandler.java [hunk 2]
```diff
@@ -132,6 +135,8 @@ public void handle(@NonNull final HandleContext context) throws HandleException
         assessAndChargeCustomFee(context, convertedOp);
```
Comment (david-bakin-sl) [inline hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenAirdropHandler.java:135]: We should have a refactor task to break this method (`.handle`) up into smaller pieces.  Its doing a lot of work.  It's all closely related, and _seems_ straightforward.  But the complexity is indicated by the fact that when it was written only _two_ unit tests were provided which does not seem adequate in any way, but it's my inference that unit testing it properly, as it stands, was too daunting.

**Review by david-bakin-sl — APPROVED** [review]
LGTM though there are suggestions.

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenAirdropHandler.java [hunk 3]
```diff
@@ -203,6 +208,33 @@ public void handle(@NonNull final HandleContext context) throws HandleException
...
+    private static void throwIfContract(TokenTransferList tokenTransferList, WritableAccountStore accountStore) {
+        var receivers = extractAllReceivers(tokenTransferList);
+
+        for (var receiver : receivers) {
+            if (receiver.alias() != null) {
+                continue;
+            }
+            var account = accountStore.get(receiver);
+            validateTrue(account != null, INVALID_ACCOUNT_ID);
+            validateFalse(account.smartContract() && account.key().contractID() != null, INVALID_TRANSACTION_BODY);
```
[the remark this answers is not in the document]
Comment (vtronkov) [reply hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenAirdropHandler.java:222]: Not sure if we have a simple term for that 🤷 I renamed it to a more descriptive name `throwIfContractWithoutAdminKey`.

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenAirdropHandler.java (hunk 2 above)
Comment (vtronkov) [reply hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenAirdropHandler.java:135]: I created an issue for that https://github.com/hashgraph/hedera-services/issues/14505 but for some reason, it got deleted 🤔 I will ask in the services channel why.

**Review by david-bakin-sl — APPROVED** [review]
LGTM

**Review by Neeharika-Sompalli — APPROVED** [review]

**Pull request merged**